parcels/interaction/base_neighbor.py

```diff
@@ -0,0 +1,126 @@
+from abc import ABC, abstractmethod
+
+import numpy as np
+from parcels.interaction.distance_utils import spherical_distance
+
+
+class BaseNeighborSearch(ABC):
+    """Base class for searching particles in the neighborhood."""
+    name = "unknown"
+
+    def __init__(self, interaction_distance, interaction_depth,
+                 max_depth=100000):
+        self.interaction_depth = interaction_depth
+        self.interaction_distance = interaction_distance
+        self.inter_dist = np.array(
+            [self.interaction_distance, self.interaction_distance,
+             self.interaction_depth]).reshape(3, 1)
+        self.max_depth = max_depth
+        self._values = None
+        self._active_mask = None
+
+    @abstractmethod
+    def find_neighbors_by_coor(self, coor):
+        '''Get the neighbors around a certain location.
+
+        :param coor: Numpy array with [lat, long, depth].
+        :returns List of particle indices.
+        '''
+        raise NotImplementedError
+
+    def find_neighbors_by_idx(self, particle_idx):
+        '''Get the neighbors around a certain particle.
+
+        Mainly useful for Structure of Array (SoA) datastructure
+
+        :param particle_idx: index of the particle (SoA).
+        :returns List of particle indices
+        '''
+        coor = self._values[:, particle_idx].reshape(3, 1)
+        return self.find_neighbors_by_coor(coor)
+
+    def update_values(self, new_values, new_active_mask=None):
+        '''Update the coordinates of the particles.
+
+        This is a default implementation simply rebuilds the structure.
+        If the rebuilding is slow, a faster implementation can be provided.
+
+        :param new_values: new coordinates of the particles.
```

Comment:
could you please add the format in which those values are expected, as in the docstring of the `find_neighbors_by_coor` function ? just for the next developer to make sure what to include in those values and how to arrange or shape them.

Reply:
I updated the docstring.